code/modules/crafting/stack_recipes/_recipe.dm

```diff
@@ -0,0 +1,224 @@
+#define REINFORCE_FORBIDDEN -1
+#define REINFORCE_ALLOWED    0
+#define REINFORCE_REQUIRED   1
+
+/*
+ * Recipe datum
+ */
+/decl/stack_recipe
+
+	var/name
+	var/result_type
+	var/req_amount     // amount of material needed for this recipe
+	var/res_amount = 1 // amount of stuff that is produced in one batch (e.g. 4 for floor tiles)
+	var/max_res_amount = 1
+	var/time = 0
+	var/one_per_turf = 0
+	var/on_floor = 0
+	var/use_material
+	var/use_reinf_material
+	var/difficulty = 1 // higher difficulty requires higher skill level to make.
+	var/apply_material_name = 1 //Whether the recipe will prepend a material name to the title - 'steel clipboard' vs 'clipboard'
+	var/set_dir_on_spawn = TRUE
+	var/expected_product_type = /obj
+	var/list/craft_stack_types
+	var/list/forbidden_craft_stack_type = list(
+		/obj/item/stack/material/ore,
+		/obj/item/stack/material/log
+	)
+
+	var/category
+	var/crafting_extra_cost_factor = 1.2
+	var/recipe_skill = SKILL_CONSTRUCTION
+
+	var/required_tool
+	var/required_reinforce_material = REINFORCE_FORBIDDEN
+	var/required_material
+
+	var/required_wall_support_value
+	var/required_integrity
+	var/required_hardness
+	var/required_max_opacity
+
+/decl/stack_recipe/validate()
+	. = ..()
```

Comment:
```suggestion
	. = ..()
	var/decl/hierarchy/skill/used_skill = GET_DECL(recipe_skill)
	if(!(difficulty in used_skill.levels)
		. += "required skill [recipe_skill] is missing skill level [isnull(difficulty) ? "NULL" : json_encode(difficulty)]"
```
I'd like to validate this since just having a band-aid solution like using LAZYACCESS doesn't solve the root cause.